app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
```

Comment:
VMware targeted refresh uses the same refresher as a full refresh, but with a subset of inventory (e.g.: only the host and datastore that the VM is on)

Not sure how feasible that is to do for RHEV, but it has the benefit of all the refreshes going through the same code paths